src/EditorFeatures/Core/InheritanceMargin/InheritanceRelationship.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.InheritanceMargin
+{
+    /// <summary>
+    /// Indicate the relationship between the member and its inheritance target
+    /// </summary>
+    [Flags]
+    internal enum InheritanceRelationship
+    {
+        /// <summary>
+        /// A default case that should not be used.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Indicate the target is implementing the member. It would be shown as I↑.
+        /// </summary>
+        Implementing = 1,
+
+        /// <summary>
+        /// Indicate the target is implemented by the member. It would be shown as I↓.
+        /// </summary>
+        Implemented = 2,
+
+        /// <summary>
+        /// Indicate the target is overriding the member. It would be shown as O↑.
+        /// </summary>
+        Overriding = 4,
+
+        /// <summary>
+        /// Indicate the target is overridden by the member. It would be shown as O↓.
```

Comment:
i don't love us encoding this.  it's unclear to me where we'll land in the future with the UI.  but this is just docs, so i'm ok with it for now.